src/analyze/accumulatedtracedata.h

```diff
@@ -172,6 +172,17 @@ struct AccumulatedTraceData
     std::vector<IpIndex> opNewIpIndices;
 
     std::vector<AllocationInfo> allocationInfos;
+
+    struct ParsingState {
+        boost::uintmax_t fileSize = -1;
```

Comment:
oops got to fix this.

Reply:
yep, and I would prefer if you use uint64_t for the fileSize

Reply:
happy to change it. That type was selected because of boost::filesystem uses that type, though that will not change any time soon, and I can't imagine overflow could ever happen here.

On that subject, if we allow/require c++17 or "::experimental" namespace we could avoid boost here.

Also, would you rather the other counts be `uint64_t` and just initalized to 0? I considered `boost::optional`s but wasn't sure if that was overkill.

I'm always torn on whether to use unsigned types. On the one hand these numbers should never really be negative, but on the other hand unsigned types can wreck havoc when they're invariably eventually used in signed math and int64 is so large the extra bit "wasted" on the sign is irrelevant especially compared to the ease of use when needing to mix signed and unsigned arithmetic. 

IIRC I think the c++ standards committee has started to move toward using signed types in more places to avoid the signed/unsigned math conflicts.

Reply:
I don't want to require c++17 yet - this makes it sometimes a bit harder to compile heaptrack for embedded systems (yes, even in 2020...). so please stick to boost for now.

regarding the signedness - :shrug: uint + 0 sounds good to me, but if you ever encounter API that returns -1 to signal an error then better stick to int + -1 or similar